### ISSUE TITLE
didn't work with latest version of web3n(1.0.0-beta.38 and later)

### ISSUE THREAD

**Issue by amazingandyyy** [vechain/thorify#72]
do 
```
const thorify = require("thorify").thorify;
const Web3 = require("web3");

const web3 = thorify(new Web3(), "http://localhost:8669");
```

```
"dependencies": {
    "thorify": "^1.0.4",
    "web3": "^1.0.0-beta.38"
  }
```

Keep throwing `Error: Missing parameter: provider` from the path of `~/node_modules/web3-core/dist/web3-core.cjs.js:163`

From web3 version 1.0.0 [docs](https://web3js.readthedocs.io/en/1.0/web3.html#id4), when we init the Web3, the parameter cannot be empty.

**Comment by amazingandyyy** [comment]
good thing to know is that 
- works
```
"dependencies": {
    "thorify": "0.3.0",
    "web3": "1.0.0-beta.34"
  }
```

**Comment by libotony** [comment]
The new version of `web3@1.0.0-beta.38` requires the parameter, I will test more with the new version

**Comment by libotony** [comment]
`beta.38` has changed a lot in the code base, I would like to wait for the code base to be stable, the code base may break the way that `thorify` extending `web3`

**Comment by amazingandyyy** [comment]
yes, the temporary plan here is to update the `thorify` documentation

**Comment by CruiseMan** [comment]
@libotony Maybe it's time to update `web3`. They have bee doing a lot of fixes https://github.com/ethereum/web3.js/ For my code i ned to have v 58 but thorify requires . 37.

**Comment by molekilla** [comment]
any update on upgrading? there is high severity issue with a web3 dependency (tar)

**Comment by libotony** [comment]
Web3 has decided to publish version 1.0 based on the codebase of `1.0.0-beta.37` for the compatibility with most packages, this is described in this [release note](https://github.com/ethereum/web3.js/releases/tag/v1.2.0). Thorify relies on the codebase of `beta.37` so I have published version 1.3.0 which works with `web3@1.x`
